core/src/main/java/hudson/model/Job.java

```diff
@@ -1210,7 +1210,8 @@ public synchronized void doConfigSubmit(StaplerRequest req,
 
             String newName = req.getParameter("name");
             final ProjectNamingStrategy namingStrategy = Jenkins.getInstance().getProjectNamingStrategy();
-            if (newName != null && !newName.equals(name)) {
+            if (newName != null && !newName.trim().equals(name.trim())) {
```

Comment:
@lanwen The original line,
```
newName != null && !newName.trim().equals(name.trim())
```
and
```
newName != null && !newName.trim().equals(name.trim())
```
are not logically equivalent.  In the former case, a null newName will result in _not_ entering the body of the if-block, but in the latter case, it would be entered under this condition.

If we use the StringUtils methods here, with the static import of #trim, the complete expression will be
```
(newName != null && !StringUtils.equals(trim(newName), trim(name)))
```
which is more verbose than the original
```
(newName != null && !newName.trim().equals(name.trim())
```
So the line would be longer and we'll need to include two additional import statements.  The original verbiage seems most concise.